datafusion/src/sql/planner.rs

```diff
@@ -1121,52 +1121,53 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
                 // then, window function
                 if let Some(window) = &function.over {
-                    if window.partition_by.is_empty() {
-                        let order_by = window
-                            .order_by
-                            .iter()
-                            .map(|e| self.order_by_to_sort_expr(e))
-                            .into_iter()
-                            .collect::<Result<Vec<_>>>()?;
-                        let window_frame = window
-                            .window_frame
-                            .as_ref()
-                            .map(|window_frame| window_frame.clone().try_into())
-                            .transpose()?;
-                        let fun = window_functions::WindowFunction::from_str(&name);
-                        if let Ok(window_functions::WindowFunction::AggregateFunction(
-                            aggregate_fun,
-                        )) = fun
-                        {
-                            return Ok(Expr::WindowFunction {
-                                fun: window_functions::WindowFunction::AggregateFunction(
-                                    aggregate_fun.clone(),
-                                ),
-                                args: self
-                                    .aggregate_fn_to_expr(&aggregate_fun, function)?,
-                                order_by,
-                                window_frame,
-                            });
-                        } else if let Ok(
-                            window_functions::WindowFunction::BuiltInWindowFunction(
+                    let partition_by = window
+                        .partition_by
+                        .iter()
+                        .map(|e| self.sql_expr_to_logical_expr(e))
+                        .into_iter()
+                        .collect::<Result<Vec<_>>>()?;
+                    let order_by = window
+                        .order_by
+                        .iter()
+                        .map(|e| self.order_by_to_sort_expr(e))
+                        .into_iter()
+                        .collect::<Result<Vec<_>>>()?;
+                    let window_frame = window
+                        .window_frame
+                        .as_ref()
+                        .map(|window_frame| window_frame.clone().try_into())
+                        .transpose()?;
+                    let fun = window_functions::WindowFunction::from_str(&name);
+                    if let Ok(window_functions::WindowFunction::AggregateFunction(
```

Comment:
I may be misreading this, but it looks like the error message may have gotten lost.

Stylistically, this might be cleaner using a `match` rather than an `if/else` chain (and the compiler will tell you if you missed a case) something like

```rust
let fun = window_functions::WindowFunction::from_str(&name)?; // note question mark
use window_functions::WindowFunction::*;

match fun {
  AggregateFunction(aggregate_fun) =>{ .. code .. },
  BuiltInWindowFunction(window_fun) =>{ .. code .. },
}
```

Perhaps?

Reply:
it's a good point. i was previously thinking maybe deferring the else the subsequent parsing part but then it's with over clause so the allowed listed of function names shall be known to be limited.

Reply:
fixed